contributed_definitions/nyaml/NXraman.yaml

```diff
@@ -0,0 +1,755 @@
+      varying temperature, pressure, electric field, ....)
+      
+      The application definition defines:
+      
+      * elements of the experimental instrument
+      * calibration information if available
+      * parameters used to tune the state of the sample
+      * sample description
+    definition:
+      doc: |
+        An application definition for ellipsometry.
+      \@version:
+        doc: |
+          Version number to identify which definition of this application
+          definition was used for this entry/data.
+      \@url:
+        doc: |
+          URL where to find further material (documentation, examples) relevant
+          to the application definition.
+      enumeration: [NXraman]
+    experiment_description:
+      doc: |
+        An optional free-text description of the experiment.
+        
+        However, details of the experiment should be defined in the specific
+        fields of this application definition rather than in this experiment
+        description.
+    experiment_type:
+      doc: |
+        Specify the type of Raman measurement.
+      enumeration: [in situ Raman spectroscopy, resonant Raman spectroscopy, non-resonant Raman spectroscopy, Raman imaging, Tip-enhanced Raman spectroscopy (TERS), Surface-enhanced Raman spectroscopy (SERS), Surface plasmon polariton enhanced Raman scattering (SPPERS), Hyper Raman spectroscopy (HRS), Stimulated Raman spectroscopy (SRS), Inverse Raman spectroscopy (IRS), Coherent anti-Stokes Raman spectroscopy (CARS)]
+    (NXinstrument):
+      doc: |
+        Properties of the Instruments used for Raman instrumeasurements.
+      company:
+        exists: optional
+        doc: |
+          Name of the company which build the instrument.
+      construction_year(NX_DATE_TIME):
+        exists: optional
+        doc: |
+          ISO8601 date when the instrument was constructed.
+          UTC offset should be specified.
+      software(NXprocess):
+        program:
+          doc: |
+            Commercial or otherwise defined given name of the program that was
+            used to generate the result file(s) with measured data and metadata.
+            This program converts the measured signals to Raman data. If
+            home written, one can provide the actual steps in the NOTE subfield
+            here.
+      incident_source_wavelength(NX_NUMBER):
+        unit: NX_LENGTH
+        doc: |
+          Which indicent wavelength was used for e.g. a laser source.
+          For multiple excitation wavelengths, please state the specific discret
+          wavelengths (i.e. for dye laser) or the specific contnious range
+          (i.e for white light laser source)
+      scattering_configuration(NX_CHAR):
+        # unit: NX_LENGTH
+        doc: |
+          Scattering configuration as defined by the porto notation by three
+          states, which are othogonal to each other. Example: z(xx)z for
+          parallel polarized backscattering configuration.
```

Comment:
Maybe a bit of introduction on the porto notation here? Not everyone might be used to this I think